metpy/calc/kinematics.py

```diff
@@ -748,3 +748,48 @@ def absolute_vorticity(u, v, dx, dy, lats, dim_order='yx'):
     f = coriolis_parameter(lats)
     relative_vorticity = vorticity(u, v, dx, dy, dim_order=dim_order)
     return relative_vorticity + f
+
+@exporter.export
+@check_units('[speed]', '[speed]', '[length]', '[length]')
+def potential_vorticity(u, v, dx, dy, lats, potential_temperature, pressure,
+                        axis=0, dim_order='yx'):
+    """Calculate the potential vorticity in a layer.
+
+        Parameters
+        ----------
+        u : (M, N) ndarray
+            x component of the wind
+        v : (M, N) ndarray
+            y component of the wind
+        dx : float
+            The grid spacing in the x-direction
+        dy : float
+            The grid spacing in the y-direction
+        lats : (M, N) ndarray
+            latitudes of the wind data
+        potential_temperature : (M, N, O) ndarray
+            potential temperature
+        pressure : (M, N, O) ndarray
+            vertical pressures
+        axis :
+            The axis number representing vertical levels (i.e. axis over which
+            the derivative of potential temperature will be computed.)
+
+        Returns
+        -------
+        (M, N) ndarray
+            potential vorticity
+
+        Notes
+        -----
+        The same formula is used for isobaric and isentropic PV analysis. Simply
+        provide winds for vorticity calculations on the desired isobaric or isentropic
+        surface. A minimum of three layers are required (one above and below the desired
+        surface).
+        """
+    avor = absolute_vorticity(u, v, dx, dy, lats, dim_order=dim_order)
+    stability = first_derivative(potential_temperature, x=pressure, axis=axis)
```

Comment:
Should probably enforce length of 3 along `axis` so folks don't give bigger arrays and get a result different than hat they expect.